### DISCUSSION TITLE
Cookie not being set in production

### DISCUSSION THREAD

**Discussion by dav1d992** [nextauthjs/next-auth#3906] [Help]
### Question 💬

I'm trying to set a refresh token in a cookie which works fine on localhost but doesn't work when deployed on to production (HTTPS). The function setting the cookie is shown below. Is there something that I've missed?  

From `[...nextauth].ts`: 
```
const setRefreshTokenCookie = (
  cookies: Cookies,
  refresh_token: string
) => {
  const date = new Date();
  const time = date.getTime();
  const expireTime = time + 24 * 60 * 60 * 1000 * 90; // 90 days
  date.setTime(expireTime);

cookies.set(
    `${useSecureCookies ? "__Host-" : ""}auth.refresh-token`,
    refresh_token,
    {
      sameSite: "strict",
      overwrite: true,
      expires: date,
      httpOnly: useSecureCookies,
    }
  );
};
```

**Notes:** 

NextAuth's [useSecureCookie](https://next-auth.js.org/configuration/options)

**localhost cookies:** 
- auth.csrf-token
- auth.refresh-token
- auth.callback-url
- auth.session-token

**production cookies:**  (contains no refresh-token)
- __Host-auth.csrf-token
- __Secure-auth.callback-url
- __Secure-auth.session-token

### How to reproduce ☕️

I think you can set a cookie as I do in the function shown above. Then deploy to another production or another secure environment.  

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by raghav275** [comment]
@dav1d992 Did you find a workaround for this? I am facing this issue as well. I am not able to set cookies in the headers in production from next auth. If you are able to kindly help

**Reply by dav1d992** [reply]
I decided to go in another direction because there's a limitation of cookie sizes on browsers (4096 bytes) and my cookie on `production` would exceed that number of bytes. 

Therefore I just set my refresh token in the JWT instead so I'm utilizing [cookie chunking](https://next-auth.js.org/configuration/options#cookies) of nextauth. Thereby I'll not have any custom cookie as shown in the above code snippet.

**Reply by max-programming** [reply]
@raghav275 Just FYI I fixed it by removing the custom `cookies` property in next auth options. If you have something like this, you can remove it and that should work
![image](https://user-images.githubusercontent.com/51731966/226096551-c2457b96-f0cf-4f94-9bda-26d73d4e2708.png)

**Comment by raghav275** [comment]
@balazsorban44 Can you please look into it?
